### ISSUE TITLE
Bug: Settings must be immutable

### ISSUE THREAD

**Issue by droganov** [nolimits4web/Swiper#1542]
If you create swiper like this:

```
componentDidMount: function(){
      const swiper = new Swiper( this.refs.container, this.props.settings );
      this.setState({
         swiper,
      });
   },
componentWillUnmount: function(){
      this.state.swiper.destroy();
   },
```

you'll be able to render react container only once, on secont attemp you'll get:
`Uncaught TypeError: Cannot read property 'nodeType' of null`

which happens to be located here:
`if (typeof params[param] === 'object' && !(params[param].nodeType || params[param] === window || params[param] === document || (typeof Dom7 !== 'undefined' && params[param] instanceof Dom7) || (typeof jQuery !== 'undefined' && params[param] instanceof jQuery))) {`

I can avoid the error like this:

```
componentDidMount: function(){
      const settings = Object.assign( {}, this.props.settings );
      const swiper = new Swiper( this.refs.container, settings );
      this.setState({
         swiper,
      });
   },
```

This leads us to idea that Swiper mutates it's settings which is a bug indeed.

**Comment by nolimits4web** [comment]
Issue is closed because of outdated/irrelevant/not actual/needed more information/inactivity.

If this issue is still actual and reproducible for latest version of Swiper, please create new issue and fill the issue template correctly:
* Clearly describe the issue including steps to reproduce when it is a bug.
* Make sure you fill in the earliest version that you know has the issue.
* Provide live link or JSFiddle/Codepen or website with issue
